src/player.cpp

```diff
@@ -5450,14 +5450,102 @@ void player::suffer()
-                mod_pain( 1 );
-            } else {
-                focus_pool --;
+            //calculate coverage for every body part
+            for( const item &i : worn ) {
+                body_part_set covered = i.get_covered_body_parts();
+                for( const body_part &bp : all_body_parts )  {
+                    if( !affected_bp.test( bp ) || !covered.test( bp ) ) {
+                        continue;
+                    }
+                    //percent of "not covered skin"
+                    float p = 1.0 - i.get_coverage() / 100.0;
+                    open_percent[bp] = open_percent[bp] * p;
+                }
+            }
+
+            body_part max_affected_bp = num_bp;
+            float max_affected_bp_percent = 0;
+            int count_affected_bp = 0;
+            for( auto &it : open_percent ) {
+                const body_part &bp = it.first;
+                float &p = it.second;
+
+                if( p < 1.0 ) {
+                    p = 0;
+                    continue;
+                }
+                ++count_affected_bp;
+                if( max_affected_bp_percent < p ) {
+                    max_affected_bp_percent = p;
+                    max_affected_bp = bp;
+                }
+            }
+            if( count_affected_bp > 0 && max_affected_bp != num_bp ) {
+                std::string bp_name;
+                switch( max_affected_bp ) {
+                    case bp_head:
+                        bp_name = _( "head" );
+                        break;
+                    case bp_torso:
+                        bp_name = _( "torso" );
+                        break;
+                    case bp_arm_l:
+                        bp_name = _( "left arm" );
+                        break;
+                    case bp_arm_r:
+                        bp_name = _( "right arm" );
+                        break;
+                    case bp_hand_l:
+                        bp_name = _( "left hand" );
+                        break;
+                    case bp_hand_r:
+                        bp_name = _( "right hand" );
+                        break;
+                    case bp_leg_l:
+                        bp_name = _( "left leg" );
+                        break;
+                    case bp_leg_r:
+                        bp_name = _( "right leg" );
+                        break;
+                    case bp_foot_l:
+                        bp_name = _( "left foot" );
+                        break;
+                    case bp_foot_r:
+                        bp_name = _( "right foot" );
+                        break;
+                    default:
+                        debugmsg( "Albino: not handled body part: %d", max_affected_bp );
+                        bp_name = _( "skin" );
+                        break;
+                }
+
+                if( count_affected_bp > 1 ) {
+                    bp_name += _( " and other body parts" );
```

Comment:
```suggestion
                    bp_name = string_format( _( "%s and other body parts" ), bp_name );
```

Reply:
You may also want to check if both arms/legs are affected and use "arms/legs" instead of "left/right arms/legs" in such case. To do this you can use `bp_aiOther` in `bodypart.h` to determine if two bodyparts are a pair.